src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -203,6 +205,9 @@ public void init(AdaptorContext context) throws Exception {
     } else {
       metadataColumns = new MetadataColumns(metadataColumnsConfig);
       log.config("metadata columns: " + metadataColumns);
+      if (includeAllColumnsAsMetadata) {
+        ignored.add("db.includeAllColumnsAsMetadata");
```

Comment:
Yes. Seems odd to me, but it's the existing behavior. The mere existence of both flags seems odd to me (essentially duplicating the SQL select list in functionality).